react/react2/react2-week1/app/components/MarsRover.js

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+
+export const MarsRover = () => {
+  const [response, setResponse] = useState([]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    fetch(
+      "https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?sol=1000&api_key=bjGJ41igTdD6HiSXcGVAaTGQ4TpvKhkl54NcWjpU"
+    )
+      .then((data) => {
+        if (!data.ok) {
+          throw new Error("Fetch failed");
+        }
+        return data.json();
+      })
+      .then((jsonData) => {
+        // Access the "photos" array from the jsonData
+        setResponse(jsonData.photos);
+      })
+      .catch((error) => {
+        setError(error.message);
+      });
+  }, []);
+
+  if (error) {
+    return <div>Error: {error}</div>;
+  }
+
+  return (
+    <>
+      <div>
+        {response.map((photo) => (
+          <div>
```

Comment:
Don't forget to use "key" prop for arrays. React, actually, trying to help you with these small (and totally normal) errors.
It is a good practice not to leave any warning behind!

<img width="807" alt="image" src="https://github.com/user-attachments/assets/fcdfd89a-1ab3-4814-b67f-b48299345031">

Reply:
Fixed , thanks for point out :)